src/frontend/Parser.y

```diff
@@ -723,6 +727,12 @@ checkConstantReassignment e = case G.expAst e of
   G.IndexAccess gId _ -> checkConstantReassignment gId
   _ -> return ()
 
+checkPointerVariable :: G.Expr -> ST.ParserMonad ()
+checkPointerVariable G.Expr {G.expType=(T.PointerT _)} = return ()
+checkPointerVariable G.Expr {G.expTok=tk} = do
+  RWS.tell [Error ("Expresion " ++ show tk ++ " must be a valid pointer") (T.position tk)]
+  return ()
```

Comment:
Thie can be simpllified as this:
```haskell
checkPointer G.Expr {G.expTok=tok} =
      RWS.tell [...]
```

Remove the `do` and the `return ()`, RWS.tell has a type like this `MonadWriter w m => w -> m ()`. The wrapped value is already `()`

Reply:
Will do